app/src/common/shared/com/igalia/wolvic/utils/EnvironmentUtils.java

```diff
@@ -60,31 +60,27 @@ public static String getEnvPath(@NonNull Context context, @NonNull String envId)
     }
 
     /**
-     * Returns a path in the external for the remote environments unzipping.
+     * Returns a path in the cache for the remote environments unzipping.
      * @param context An activity context.
      * @param envId The environment id. This maps to the Remote properties JSON "value" environment property.
      * @return The location of the environment in the devices memory.
      */
     @Nullable
     public static String getExternalEnvPath(@NonNull Context context, @NonNull String envId) {
-        File outputFolder = context.getExternalFilesDir(ENVS_FOLDER);
-        if (outputFolder != null) {
-            outputFolder = new File(outputFolder, envId);
-            if (!outputFolder.exists()) {
-                if (outputFolder.mkdirs()) {
-                    return outputFolder.getAbsolutePath();
-
-                } else {
-                    return null;
-                }
+        File outputFolder = new File(context.getCacheDir().getAbsolutePath() + "/" + ENVS_FOLDER);
+        outputFolder = new File(outputFolder, envId);
```

Comment:
Why we need to assign a new value to the outputFolder ? Couldn't we just use the envId in the initialization value already ?

Reply:
Done! I was focusing on the change and didn't try to clean up the code.